ttnn/cpp/ttnn/operations/eltwise/binary/binary_compiler_interface.hpp

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <exception>
+#include <functional>
+
+#include "third_party/json/json.hpp"
+#include "tt_metal/common/logger.hpp"
+#include "ttnn/compiler_interface/compiler_interface.hpp"
+#include "ttnn/graph/graph_processor.hpp"
+#include "ttnn/graph/graph_trace_utils.hpp"
+
+namespace ttnn::compiler_interface {
+
+template <auto BinaryFunction>
+QueryResponse binary_op_constraints(
+    Device* device, const TensorSpec& input_a, const TensorSpec& input_b, const TensorSpec& output) {
```

Comment:
How important is it that this api accepts a `TensorSpec` and not a `Tensor`?

I first thought that its best for it to be a Tensor, because it'd allow to rely on our existing decoration fascilities.
But after further look with Stas, I am worried that creation of an input tensor here messes up with allocator, even if we rely on a virtual allocator state. It is very possible that this tensor was created at a different time and occupies a different slot in memory.

Reply:
Today we dont have a mockup device nor a virtual allocator. We are running no dispatch mode as is. 

Given this work is between two repos, I felt playing safe would be to create Tensor and test on tt-metal end, and have guaranteed behavior. 

I understand your point of view, I can move to Tensor if that's your preference.